lib/src/ultrahdr_api.cpp

```diff
@@ -525,6 +525,21 @@ UHDR_EXTERN uhdr_error_info_t uhdr_enc_set_gainmap_scale_factor(uhdr_codec_priva
   return status;
 }
 
+UHDR_EXTERN uhdr_error_info_t uhdr_enc_set_gainmap_gamma(uhdr_codec_private_t* enc,
+                                                         float gamma) {
+  uhdr_error_info_t status = g_no_error;
+  uhdr_encoder_private* handle = dynamic_cast<uhdr_encoder_private*>(enc);
+  if (handle == nullptr) {
+    status.error_code = UHDR_CODEC_INVALID_PARAM;
+    status.has_detail = 1;
+    snprintf(status.detail, sizeof status.detail, "received nullptr for uhdr codec instance");
+    return status;
+  }
+
```

Comment:
error checks for gamma range are required. 
check required if context has moved to running state from init (m_sailed).